### PR TITLE
Fixed BB-169 Identifier.update doesn't work properly.

### PR DESCRIPTION
**Pull request by s17k**
[ticket](http://tickets.musicbrainz.org/browse/BB-169)

### CODE REVIEW

File: bbschema/entity.py [hunk 1]
```diff
@@ -343,10 +343,11 @@ def update(self, identifier_json):
 
         if 'value' in identifier_json:
             new.value = identifier_json['value']
-        if 'identifier_type_id' in identifier_json:
+
+        if 'identifier_type' in identifier_json:
             new.identifier_type_id =\
-                identifier_json.get('identifier_type', {}).\
-                get('identifier_type_id')
+                identifier_json.get('identifier_type').\
```
Comment (BenOckmore) [inline bbschema/entity.py]: This won't work - get('identifier_type') will either return an IdentifierType or None. None has no attribute "get", so it'll raise an exception in those cases. Should just be left as is, I think.

File: bbschema/entity.py [hunk 2]
```diff
@@ -343,10 +343,11 @@ def update(self, identifier_json):
...
+                get('identifier_type_id', None)
```
Comment (BenOckmore) [inline bbschema/entity.py]: No point in this, as this is the default behaviour.

File: bbschema/entity.py [hunk 3]
```diff
@@ -343,10 +343,11 @@ def update(self, identifier_json):
 
         if 'value' in identifier_json:
             new.value = identifier_json['value']
-        if 'identifier_type_id' in identifier_json:
+
+        if 'identifier_type' in identifier_json:
```
Comment (BenOckmore) [inline bbschema/entity.py:347]: OK to change this, but I think we have a huge inconsistency across the WS about whether to accept objects containing an ID field, or whether to just accepts a flattened ID field.

File: bbschema/entity.py (hunk 1 above)
Comment (s17k) [inline bbschema/entity.py]: I don't see a point in using the second argument of get.
It s already checked here in the previous line - that's why I erased it.

``` python
        if 'identifier_type' in identifier_json:
```

get('identifier_type') will either return a dict with IdentifierType or None (or anything else which is in identifier_json['identifier_type'],  but setting {} as default in get doesn't change anything

File: bbschema/entity.py [hunk 4]
```diff
@@ -343,9 +343,10 @@ def update(self, identifier_json):
...
             new.identifier_type_id =\
-                identifier_json.get('identifier_type', {}).\
+                identifier_json.get('identifier_type').\
```
Comment (BenOckmore) [inline bbschema/entity.py]: As you pointed out, we check for this before trying to access it, so we can just do:

```
identifier_json['identifier_type'].get('identifier_type_id')
```

**Pull request merged**